Sources/Mistica/Components/Sheet/SheetTransition/SheetPresentationController.swift

```diff
@@ -47,242 +84,47 @@ final class SheetPresentationController: UIPresentationController {
-    }
-
-    // MARK: - UI Elements
-
-    /// The view displayed behind the presented controller.
-    private lazy var overlayView: UIView = {
-        let view = UIView()
-        view.alpha = .zero
-        view.backgroundColor = .backgroundOverlay
-        view.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(tappedBackgroundView)))
-        return view
-    }()
-
-    /// The view displaying a handle on the presented view.
-    private let handleView: UIView = {
-        let view = UIView()
-        view.backgroundColor = .control
-        view.frame.size = CGSize(width: 40, height: 4)
-        return view
-    }()
-
-    /// The pan gesture used to drag and interactively dismiss the sheet.
-    private lazy var panGesture = UIPanGestureRecognizer(target: self, action: #selector(pannedPresentedView))
-
-    // MARK: - Methods
-
-    override func presentationTransitionWillBegin() {
-        super.presentationTransitionWillBegin()
-
-        containerView?.addSubview(overlayView)
-
-        presentedView?.addSubview(handleView)
-
-        overlayView.fade(toAlpha: 1.0, duration: 0.25)
-
-        presentedViewController.transitionCoordinator?.animate(alongsideTransition: { [weak self] _ in
-            guard let self = self else {
-                return
-            }
-
-            self.presentedView?.layer.cornerRadius = self.cornerRadius
-        })
-    }
-
-    override func containerViewDidLayoutSubviews() {
-        super.containerViewDidLayoutSubviews()
-
-        layoutAccessoryViews()
-
-        guard let presentedView = presentedView else {
-            return
-        }
-
-        setupPresentedViewInteraction()
-
-        if #available(iOS 13.0, *) {
-            presentedView.layer.cornerCurve = .continuous
-        }
-        presentedView.layer.maskedCorners = [.layerMinXMinYCorner, .layerMaxXMinYCorner]
-
-        presentedViewController.additionalSafeAreaInsets.top = handleView.frame.maxY
-    }
-
-    override func dismissalTransitionWillBegin() {
-        super.dismissalTransitionWillBegin()
-
-        presentedViewController.transitionCoordinator?.animate(alongsideTransition: { [weak self] _ in
-            guard let self = self else {
-                return
-            }
-
-            self.presentedView?.layer.cornerRadius = .zero
-            self.overlayView.fade(toAlpha: 0.0, duration: 0.25)
-        })
-    }
 }
 
 private extension SheetPresentationController {
-    // MARK: - Private Helpers
-
-    /// Lays out the accessory views of the presentation.
-    private func layoutAccessoryViews() {
-        guard let containerView = containerView else {
-            return
-        }
-
-        overlayView.frame = containerView.bounds
-
-        guard let presentedView = presentedView else {
-            return
-        }
+    var maxTopOffset: CGFloat {
+        let safeTopInset: CGFloat
+        safeTopInset = containerView?.safeAreaInsets.top ?? 0
```

Comment:
```suggestion
        let safeTopInset: CGFloat = containerView?.safeAreaInsets.top ?? 0
```

I am not sure if we need the type here.

Reply:
Yes, many lines. Fixed!